### ISSUE TITLE
SSH Access w/o Port Mapping?

### ISSUE THREAD

**Issue by chrisRidgers** [phusion/baseimage-docker#362]
Hi Guys,

Just built a new box using this as my Dockerfile:

```
FROM phusion/baseimage

CMD ["/sbin/my_init"]

RUN apt-get update && apt-get upgrade -y -o Dpkg::Options::="--force-confold"

RUN rm -f /etc/service/sshd/down

RUN /etc/my_init.d/00_regen_ssh_host_keys.sh

ADD id_rsa.pub /tmp/id_rsa.pub

RUN cat /tmp/id_rsa.pub >> /root/.ssh/authorized_keys && rm -f /tmp/id_rsa.pub

RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

When I run my container using: `docker run -d --rm -t -i <image>`

I'm unable to login via ssh.  `ssh root@<ip>` hangs.

However if i run: `docker run -d --rm -t -i -p 2022:22 <image>`

I can then ssh with: `ssh -p2022 root@localhost`

Is it the docs are out of date?  Or is there another reason I can't login without declaring port mapping?

**Comment by kingdonb** [comment]
If you read carefully the section(s) on connecting to containers using SSH describes how to identify the IP address of the container with `<ID>`, repeating this incantation several times:

```
docker inspect -f "{{ .NetworkSettings.IPAddress }}" <ID>
```

This container address that is returned by the inspect command is a docker private address that can only be accessed from a local user on the docker host (or from another host in the swarm, if that's how docker swarm handles networking, not exactly sure.  All Docker orchestration systems have different mechanisms for container networking and shared private container addressing.)

If you want to access SSH from a different host elsewhere on the network, then port mapping as you've used it here is the way to go.  This will expose your port on the real IP network.  If you only want to SSH into the container from a user on the same docker host, you can use docker inspect as shown to find the address, and that's where your containerized SSH server will be listening on port 22 (not on localhost.)

**Comment by kingdonb** [comment]
@chrisRidgers If that doesn't answer your question, it is also possible that something broke.  It's not clear from your post which IP you're using to try the SSH without the port mapping.

**Comment by chrisRidgers** [comment]
yeh I followed those docks to get the IP address.  However sshing at that address failed to connect.

If by docker host you mean the machine running 'Docker for Mac' then thats correct too.

So I'm on my physical machine, with Docker for Mac running, and have successfully spun up my container.  I can read the IP address as documented, but then not login.  However when I instead run the machine explicitly exposing the ports: I can then access them over localhost via whatever port mapping I've configured.

I'm still fairly new at Docker... but the last time I was using it the process revolved around boot2docker.  I'm just not sure if what I'm doing is wrong, 'Docker for Mac' works differently, or the docs don't apply / haven't been updated to reflect any changes.
